apacheconfig/wloader.py

```diff
@@ -30,98 +28,149 @@ class Node():
 
     @abc.abstractmethod
     def __str__(self):
-        raise NotImplementedError()
+        """Writes this node to a raw string. To get more metadata about the
+        object, use ``repr``.
+        """
+        pass
+
+    @abc.abstractmethod
+    def __repr__(self):
+        """Returns a string containing object metadata."""
+        pass
 
     @property
-    def type(self):
+    def parser_type(self):
         """A typestring as defined by the apacheconfig parser.
 
-        A single *Node class can have multiple possible values for this. For
-        instance, ItemNode is a generic representation of any single directive
-        but has a couple different possible `type`s, including `comment`,
-        `statement`, and `include`-- depending on which the caller may
-        decide to treat ItemNode differently.
+        The possible values for this are:
+
+        :class:`apacheconfig.ItemNode`: ``comment``, ``statement``,
+        ``include``, ``includeoptional``
+
+        :returns: The typestring (as defined by the apacheconfig parser) for
```

Comment:
Would it make us suffering less if we'd use Napoleon for the docstrings? Vanilla Sphinx parameters syntax seems somewhat verbose and demanding...

FWIW, Napoleon offers two docstring styles to choose from Google and NumPy.

WDYT?

Reply:
Great suggestion! It's much easier and more readable in code, too-- I gave it a shot in the most recent commit. The branch `sydneyli/apacheconfig:merged-writable-loader-item-sphinx` shows the doc build passing with the napoleon extension.